celery/canvas.py

```diff
@@ -1452,6 +1471,7 @@ def run(self, header, body, partial_args, app=None, interval=None,
         # we execute the body manually here.
         else:
             body.delay([])
+            header_result = self.app.GroupResult(*header_result_args)
```

Comment:
Was this a different bug?

Reply:
Creating a GroupResult will drain the header generator, so I had to refactor it out of `apply_chord`. This line is the last piece of that refactoring.

Reply:
Got it.